connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -2413,23 +2402,10 @@ ble_error_t Gap::startAdvertising(
             _pal_gap.set_advertising_set_random_address(handle, *random_address);
         }
 
-        error = _pal_gap.extended_advertising_enable(
-            /* enable */ true,
-            /* number of advertising sets */ 1,
-            &handle,
-            maxDuration.storage(),
-            &maxEvents
-        );
-
-        if (error) {
-            return error;
-        }
-
-        if (maxDuration.value() || maxEvents) {
-            _interruptible_sets.clear(handle);
-        } else {
-            _interruptible_sets.set(handle);
-        }
+        _event_queue.post([this, handle, maxDuration, maxEvents] {
```

Comment:
Why not execute this immediately ? Queuing the advertising into the data structure shouldn't interfere with the process and `evaluate_advertising_enable_queue` looks if `_advertising_enable_pending` is set or not.

Reply:
this is the avoid race conditions with events